msu/utilities/log.nut

```diff
@@ -0,0 +1,87 @@
+::MSU.Log <- {
+	// maxLen is the maximum length of an array/table whose elements will be displayed
+	// maxDepth is the maximum depth at which arrays/tables elements will be displayed
+	// advanced allows the ID of the object to be displayed to identify different/identical objects
+	function printStackTrace( _maxDepth = 0, _maxLen = 10, _advanced = false )
+	{
+		local count = 2;
+		local string = "";
+		while (getstackinfos(count) != null)
+		{
+			local line = getstackinfos(count++);
+			string += "Function:\t\t";
+
+			if (line.func != "unknown")
+			{
+				string += line.func + " ";
+			}
+
+			string += "-> " + line.src + " : " + line.line + "\nVariables:\t\t";
+
+			foreach (key, value in line.locals)
+			{
+				string += this.getLocalString(key, value, _maxLen, _maxDepth, _advanced);
+			}
+			string = string.slice(0, string.len() - 2);
+			string += "\n";
+		}
+		::logInfo(string);
+	}
+
+	function printData( _data, _maxDepth = 1, _advanced = false )
+	{
+		local maxLen = 1;
+		if (typeof _data == "array" || typeof _data == "table"){
+			maxLen = _data.len();
+		}
+		::logInfo(this.getLocalString("Printing Data", _data, maxLen, _maxDepth, _advanced));
+	}
+
+	function getLocalString( _key, _value, _maxLen, _depth, _advanced, _isArray = false )
```

Comment:
I still wanna say I hate this function and I think it's really ugly but idk how to make it cleaner

Reply:
Looks ok to me. Can't say it's not ugly, but it's ok.